week-4/InClass/E-arrays-of-objects/exercise-3.js

```diff
@@ -54,32 +54,51 @@ DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
 
-
 let restaurantFinderApplication = {
-    applicationName: "Restaurant Finder",
-    applicationVersion: "1.0",
-    restaurants: restaurants,
-    findAvailableRestaurants: function (numberOfPeople) {
-        // Complete here
-    },
-    findRestaurantServingDish: function (dishName) {
-        // Complete here
-    },
-    countNumberOfRestaurantsInArea: function (area) {
-        // Complete here
-    }
+  applicationName: "Restaurant Finder",
+  applicationVersion: "1.0",
+  restaurants: restaurants,
+  findAvailableRestaurants: function(numberOfPeople) {
+    return restaurants
+      .filter(
+        numPeople =>
+          numPeople.totalSeats - numPeople.numberOfCustomers >= numberOfPeople
+      )
+      .map(resName => resName.name);
+  },
```

Comment:
👏 Excellent! Just be mindful of the parameter names again. `numPeople` and `resName` don't make sense as those arrow function will both receive a `restaurant` as parameters.